### ISSUE TITLE
Support entering math using MathType

### ISSUE THREAD

**Issue by ronaldtse** [glossarist/glossarist-desktop#47]
This was a request from Peter Godwin of IEC.

It should be possible to integrate their "Offline JavaScript JDK" in Glossarist:
* https://www.npmjs.com/package/@wiris/mathtype-integration-js-dev
* https://github.com/wiris/mathtype-integration-js-dev/

The text editing module of Glossarist currently allows entering AsciiMath, but does not provide a WYSIWYG editor for entering math formulas.

This work is to integrate JavaScript WIRIS MathType into Glossarist as an option for people to use. 

WIRIS MathType is a licensed software, and therefore cannot be "freely available" to all Glossarist users. For organizations that have licensed MathType, they should be able to "turn on" MathType in Glossarist, and use MathType to enter math formulas in the Glossarist text editing module. (e.g. in the Settings pane).

Glossarist stores math in MathML; MathType has an ability to store and read math in MathML.

You should investigate other MathType integrations for reference, i.e. plugins to Moodle, CKEditor, TinyMCE.

WIRIS MathType Web: https://docs.wiris.com/en/mathtype/mathtype_web/start

**Comment by bravedragon331** [comment]
Which text editor do you want to change into WIRIS editor?
Destination?
![image](https://user-images.githubusercontent.com/32223911/90846685-44af5a00-e39b-11ea-8c98-edc7c45aa16b.png)
Example?
![image](https://user-images.githubusercontent.com/32223911/90846717-63adec00-e39b-11ea-9148-40936ff72882.png)

**Comment by ronaldtse** [comment]
Both Definition and EXAMPLE/NOTEs should support entering math equations using MathType.

**Comment by strogonoff** [comment]
Here are the fields where math is supported: https://github.com/glossarist/glossarist-desktop/search?l=TSX&q=MathJax

* Designation expression
* Definition

We will add that support to notes and examples as well.

**Comment by strogonoff** [comment]
Accidental close.

To see where we support MathJax currently, see uses of `MathJax.Text` in .tsx files. @ronaldtse is right but we also have math in designations.

**Comment by strogonoff** [comment]
@bravedragon331 Essentially, as a deliverable, we could settle on a React component that “wraps” another React component, and adds MathType features to the component it wraps. What do you think?

In this scenario, your component will _not_ be wrapping textarea directly. If that turns out to be a big issue and you need to wrap a textarea or input node directly, or if you cannot wrap inputs because of MathType limitations, please let us know!

**Comment by strogonoff** [comment]
(@ronaldtse for us it means we could use a flag whether MathType is to be enabled or not; if flag is on we would dynamically import() and render the new component, if off we would render what we render now. We should be able [dynamically install](https://www.npmjs.com/package/live-plugin-manager) the new component as a dependency when user with license launches the app, avoiding inclusion of proprietary code in the distributed binary.)

**Comment by ronaldtse** [comment]
Agree, thanks @strogonoff!

**Comment by strogonoff** [comment]
Can push the component to this repo: https://github.com/glossarist/mathtype-input

**Comment by bravedragon331** [comment]
https://github.com/bravedragon331/glossarist-desktop

I don't have access to your repository, so I have pushed my work here...
anyway, the newly created component is WirisMathType, WirisScriptInsert from widgets.tsx
Please run this app and check how it looks like.

thank you

**Comment by ronaldtse** [comment]
@bravedragon331 Thanks. I noticed that your repository is not a fork. Can you make it a fork and place your commits on top so we can make a Pull Request?

**Comment by strogonoff** [comment]
@bravedragon331 @ronaldtse No need to fork the repo. The deliverable is supposed to be an independent component not importing anything from Glossarist or Couomb.

**Comment by strogonoff** [comment]
> Can push the component to this repo: https://github.com/glossarist/mathtype-input

Please push your component to this repo

**Comment by bravedragon331** [comment]
> https://github.com/glossarist/mathtype-input

Please invite me to this repo

**Comment by ronaldtse** [comment]
@bravedragon331 you have access to that repo now.

**Comment by bravedragon331** [comment]
https://github.com/glossarist/mathtype-input/commit/6c43ae4f73dd4e9c99bbb655b94ec92cf17655f9
This is math-type component which is wrapped wiris editor

https://github.com/bravedragon331/glossarist-desktop/commit/feae278157a2cc3ea75c5cbd015da4d63199e2bc
This is component usage for glossarist-desktop

**Comment by strogonoff** [comment]
Thanks! Taking a look, I believe https://github.com/glossarist/mathtype-input/blob/master/src/components/widget.tsx is where the core component is

**Comment by bravedragon331** [comment]
> Thanks! Taking a look, I believe https://github.com/glossarist/mathtype-input/blob/master/src/components/widget.tsx is where the core component is

yes

**Comment by bravedragon331** [comment]
@strogonoff  did you take a look?

**Comment by bravedragon331** [comment]
hi @ronaldtse 
Could you please email to bravedragon632@gmail.com?
i have something to discuss with you.
thank you.

**Comment by strogonoff** [comment]
> @strogonoff did you take a look?

@bravedragon331 I did. I am looking into improving math type integration in a way that works in smaller single-line fields like designations, and making fields support arbitrary surrounding text (possibly RTL for Arabic).

**Comment by strogonoff** [comment]
There may be a further task along this topic in a couple days, will let you know ASAP

**Comment by bravedragon331** [comment]
yes, that's possible. I have checked code before and I can make props, 
:)

**Comment by bravedragon331** [comment]
@ronaldtse 
Did you check email? I am still waiting your answer.
Thank you.

**Comment by bravedragon331** [comment]
@ronaldtse , how is it going? I am still waiting your answer on the payment of MathType.
I didn't get paid yet...

**Comment by bravedragon331** [comment]
@ronaldtse , Hi, Ron. I have sent an email.
It would be appreciated if you reply soon.

**Comment by bravedragon331** [comment]
@ronaldtse  Hi, Ronald. I have sent an invoice. Could you check please?

**Comment by ronaldtse** [comment]
@bravedragon331 we have already paid you. Did you not see? If that's the case we need to revoke the payment.

**Comment by bravedragon331** [comment]
let me check and let you kknow

**Comment by bravedragon331** [comment]
Yes, it is arrived correctly. 
I will remove above comments and this one in few minutes.
And I am waiting new task.
Thank you.
